calyx-ir/src/structure.rs

```diff
@@ -557,19 +570,27 @@ pub struct StaticGroup {
 
     /// Attributes for this group.
     pub attributes: Attributes,
+
+    /// Latency of static group
+    pub latency: u64,
 }
 
 ///implement the StaticGroup struct
 impl StaticGroup {
-    pub fn new(name: Id) -> Self {
+    pub fn new(name: Id, latency: u64) -> Self {
         Self {
             name,
             assignments: vec![],
             holes: smallvec![],
             attributes: Attributes::default(),
+            latency,
```

Comment:
Yeah, I think you're right. For `go` holes, I realize that we probably don't need them at all. For `done` holes, this relates do the point you made about the redundancy of `static group g<4>` and `g[done] = %4 ? 1'd1`, since they're really saying the same thing. 
To your point on redundancy, I currently make the `well-formed` pass check that the assignment to `g`'s done hole is exactly `g[done] = latency ? 1'd1`. 

It might be helpful so the group has no done holes: instead, it just has `static group g<4>` to tell the static latency, and there is no assignment at all that says `g[done] = %4 ? 1'd1`.